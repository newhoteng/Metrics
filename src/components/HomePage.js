import AirQDataContainer from './AirQDataContainer';
import HeroSection from './HeroSection';

const Homepage = () => {
  const image = '';
  const mainText = 'selected cities';
  const secondaryText = 'air quality index';
  return (
    <main>
      <HeroSection image={image} mainText={mainText} secondaryText={secondaryText} />
      <AirQDataContainer />
    </main>
  );
};

export default Homepage;
