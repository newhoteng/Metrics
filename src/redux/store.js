import { configureStore } from '@reduxjs/toolkit';
import currentAQIsReducer from './current/currentSlice';

export const store = configureStore({
  reducer: {
    currentAQIs: currentAQIsReducer,
  },
});

export default store;
