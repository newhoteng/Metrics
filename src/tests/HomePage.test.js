// import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
// import { BrowserRouter } from 'react-router-dom';
// import configureStore from 'redux-mock-store';
// import thunk from 'redux-thunk';
import { store } from '../redux/store';
// import { getCurrentAQIs } from './redux/current/currentSlice';

// Component imports
import HomePage from '../components/HomePage';

// const mockStore = configureStore([thunk]);

// jest.setTimeout(15000);

// describe('Current component', () => {
//   test('renders correctly', async () => {
//     const store = mockStore({
//       currentAQIs: {
//         currentAQIs: [],
//       },
//     });

//     // Dispatch the asynchronous action
//     await store.dispatch(getCurrentAQIs());

//     const { container } = render(
//       <Provider store={store}>
//         <BrowserRouter>
//           <Current />
//         </BrowserRouter>
//       </Provider>,
//     );

//     expect(container).toMatchSnapshot();
//   });
// });


it('home page renders correctly', () => {
  const HomePageComponent = renderer
    .create(
      <Provider store={store}>
        <HomePage page="/">Home Page</HomePage>
      </Provider>,
    )
    .toJSON();
  expect(HomePageComponent).toMatchSnapshot();
});

// it('renders Current component', () => {
//   render(
//     <Provider store={store}>
//       <Current />
//     </Provider>,
//   );
// });