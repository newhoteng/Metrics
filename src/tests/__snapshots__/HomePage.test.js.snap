// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page renders correctly 1`] = `
<main
  className="main"
>
  <section
    className="hero"
  >
    <div />
    <div>
      <p>
        SELECTED CITIES
      </p>
      <p>
        air quality index
      </p>
    </div>
  </section>
  <section>
    <div
      className="banner"
    >
      INDEX BY CITY
    </div>
    <div
      className="countriesContainer"
    />
  </section>
</main>
`;
